conda_build_all/version_matrix.py

```diff
@@ -72,6 +73,41 @@ def conda_special_versions(meta, index, version_matrix=None):
             yield case
 
 
+def parse_specifications(requirements):
+    """
+    Parse a list of specifications, turning multi-line specifications into
+    a single specification.
+    """
+    requirement_specs = defaultdict(list)
+    # Generate a list of requirements for each spec name to ensure that
+    # multi-line specs are handled.
+    for spec in requirements:
+        spec_details = spec.split()
+        if len(spec_details) == 2:
+            # Package name and version spec were given, append the
+            # version spec.
+            requirement_specs[MatchSpec(spec).name].append(spec_details[1])
+        elif len(spec_details) == 1:
+            # Only package name given (e.g. 'numpy'), so an empty list works.
+            requirement_specs[MatchSpec(spec).name] = []
```

Comment:
Doesn't this clear an existing spec, i.e.

```
requirements:
  build:
    - numpy >=1.9
    - numpy
```

Would result in requirements specs: ```{'numpy': []}```?